README.md

```diff
@@ -1,15 +1,103 @@
 
-A demo web app for MAX using the Image Caption Generator model
+Every day 2.5 quintillion bytes of data are created, based on an [IBM study](https://public.dhe.ibm.com/common/ssi/ecm/wr/en/wrl12345usen/watson-customer-engagement-watson-marketing-wr-other-papers-and-reports-wrl12345usen-20170719.pdf). A lot of that data is unstructured data,
+such as large texts, audio recordings, and images. In order to do something useful
+with the data, we must first convert it to structured data.
 
-## Before Starting the Web App
+In this Code Pattern we will use one of the models from the [Model Asset Exchange](https://developer.ibm.com/code/exchanges/models/),
+an exchange where developers can find and experiment with open source deep learning
+models. Specifically we will be using the [Image Caption Generator](https://github.com/IBM/MAX-Image-Caption-Generator)
+to create a web application that will caption images and allow the user to filter through
+images based image content. The web application provides an interactive user interface
+backed by a lightweight python server using Tornado. The server takes in images via the
+UI and sends them to a REST end point for the model and displays the generated
+captions on the UI. The model's REST endpoint is set up using the docker image
+provided on MAX. The Web UI displays the generated captions for each image as well
+as an interactive word cloud to filter images based on their caption.
 
-Before starting this web app you must setup the MAX Image Caption Generator REST endpoint by following the README at:
+When the reader has completed this Code Pattern, they will understand how to:
 
-[MAX Image Caption Generator GitHub](https://github.com/IBM/MAX-Image-Caption-Generator)
+* Build a Docker image of the Image Caption Generator MAX Model
+* Deploy a deep learning model with a REST endpoint
+* Generate captions for an image using the MAX Model's REST API
+* Run a web application that using the model's REST API
+
+
+# Steps
+
+## Setting up the MAX Model
+
+Note: The set of instructions in this section are a modified version of the one found on the [Image Caption Generator Project Page](https://github.com/IBM/MAX-Image-Caption-Generator)
+
+### 1. Build the Model
+
+Clone the Image Caption Generator model locally. In a terminal, run the following command:
+
+```
+$ git clone https://github.com/IBM/MAX-Image-Caption-Generator.git
+```
+
+Change directory into the repository base folder:
+
+```
+$ cd MAX-Image-Caption-Generator
+```
+
+To build the docker image locally, run:
+
+```
+$ docker build -t max-im2txt .
+```
+
+All required model assets will be downloaded during the build process. _Note_ that currently this docker image is CPU only (we will add support for GPU images later).
+
+### 2. Deploy the Model
+
+To run the docker image, which automatically starts the model serving API, run:
+
+```
+docker run -it -p 5000:5000 -p 8088:8088 --name max-im2txt max-im2txt
```

Comment:
Note here: by running this docker command the web app must also run with docker and cannot be run locally. This is because this command takes over the default port of 8088 blocking the web app from using it without being run inside the docker container. 

So we either need to switch this to use the original command or change the Starting the Web App section to use docker. Personally I'd rather run the web app outside docker as the default steps, but it be simpler to do everything in docker. 

@frreiss in your opinion as a "user" would you rather do everything in docker or do the minimal in docker?

Reply:
It's normal to have multiple supported mechanisms (like cloud foundry or docker or run locally), just be sure to document each sufficiently. see https://github.com/IBM/kafka-streaming-click-analysis#steps as an example.

Reply:
@djalova I think we should do two separate steps for "normal" then for single docker container. I think we can open another pr for that though so for now lets just change this line back to 
```
docker run -it -p 5000:5000 max-im2txt
```

Reply:
Also should we include `-d` in our docker commands in this doc? I find that I tend to edit every command to add it anyway

Reply:
I included both commands with an explanation. I think I'll keep it running in the foreground in case the user doesn't know how to stop it.